cmd/build-waiter/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+var (
+	mode     = flag.String("mode", "wait", "one of: wait or copy")
+	to       = flag.String("to", "", "where to copy this binary")
+	waitFile = flag.String("wait-file", "", "file to wait on")
+	doneFile = flag.String("done-file", "", "file to write on completion")
+	execute  = flag.String("execute", "", "What to run after waiting")
+	errFile  = flag.String("error-file", "", "shared error file")
+)
+
+func main() {
+	flag.Parse()
+
+	switch *mode {
+	case "wait":
+		if *errFile != "" {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			go watchForErrors(ctx, *errFile)
+		}
+
+		err := wait(*waitFile, *doneFile, *execute)
+		if err != nil {
+			exitWithError(err.Error(), *errFile)
+		}
+
+	case "copy":
+		err := copy(*to)
+		if err != nil {
+			exitWithError(err.Error(), *errFile)
+		}
+	}
+}
+
+func waitForStep(waitPath string) error {
+	tickerChan := time.NewTicker(time.Second)
+	defer tickerChan.Stop()
+
+	for range tickerChan.C {
```

Comment:
can this possibly wait forever?

Reply:
ya I originally had a timeout but removed it. I figured that currently a build can execute forever and the solution is to provide an active deadline seconds when configuring the image which would still work in this case.